pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RelToStageConverter.java

```diff
@@ -151,34 +152,39 @@ private static DataSchema toDataSchema(RelDataType rowType) {
   }
 
   public static DataSchema.ColumnDataType convertToColumnDataType(RelDataType relDataType) {
-    switch (relDataType.getSqlTypeName()) {
+    SqlTypeName sqlTypeName = relDataType.getSqlTypeName();
+    boolean isArray = sqlTypeName == SqlTypeName.ARRAY;
```

Comment:
Here we are equating our current support of MV with `SqlTypeName.ARRAY`. Is `SqlTypeName.ARRAY` extendible in future to help handle complex LIST (list that may not necessarily have primitives only) ?

Reply:
From what I've read from here: https://calcite.apache.org/javadocAggregate/org/apache/calcite/sql/type/SqlTypeName.html
Seems Array is the type.

For future extensibility, we can make this call recursive to construct the recursive DataType for Complex/nest data structure.